### ISSUE TITLE
What is private repositories? 

### ISSUE THREAD

**Issue by SergeyDjam** [pulumi/pulumi#14541]
### What happened?

```
+ export GO111MODULE=on
+ GO111MODULE=on
+ export GOPROXY=direct
+ GOPROXY=direct
+ pushd pkg
+ go mod download
go: gocloud.dev@v0.27.0 requires
	contrib.go.opencensus.io/exporter/stackdriver@v0.13.13 requires
	github.com/prometheus/prometheus@v0.35.0 requires
	github.com/containerd/containerd@v1.6.1 requires
	github.com/containerd/aufs@v1.0.0 requires
	github.com/containerd/containerd@v1.5.0-beta.3 requires
	github.com/Microsoft/hcsshim@v0.8.15 requires
	github.com/containerd/containerd@v1.5.0-beta.1 requires
	github.com/Microsoft/hcsshim/test@v0.0.0-20201218223536-d3e5debf77da requires
	github.com/docker/distribution@v0.0.0-20190905152932-14b96e55d84c requires
	github.com/mitchellh/osext@v0.0.0-20151018003038-5e2d6d41470f: invalid version: git ls-remote -q origin in /builddir/go/pkg/mod/cache/vcs/94ed57c5b21c953d93b47487113db43a5c9b69fd990329ec70dc77348c4dd443: exit status 128:
	fatal: could not read Username for 'https://github.com': terminal prompts disabled
Confirm the import path was entered correctly.
If this is a private repository, see https://golang.org/doc/faq#git_https for additional information.
````

### Example

download v3.93.0 tar.gz on github
unpack

pushd pkg
go mod download
go build "-buildmode=pie"  -ldflags "-s -w -X github.com/pulumi/pulumi/pkg/v3/version.Version=%{version}" -o ../bin/%{name} ./cmd/pulumi
popd

### Output of `pulumi about`

v3.93.0/tar.gz

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by Frassle** [comment]
It looks like github.com/mitchellh/osext has been made private / deleted. 
I imagine this might force an update of go cloud.dev as the latest versions don't depend on that module it seems.

**Comment by Frassle** [comment]
I think the go module caches at proxy.golang.org still have this cached, so if you don't set GOPROXY=direct it should work.

**Comment by SergeyDjam** [comment]
Whithout GOPROXY
```
+ export GO111MODULE=on
+ GO111MODULE=on
+ pushd pkg
+ go mod download
~/build/BUILD/pulumi-3.93.0/pkg ~/build/BUILD/pulumi-3.93.0
go: cloud.google.com/go@v0.110.4: GOPROXY list is not the empty string, but contains no entries
```

**Comment by Frassle** [comment]
I suspect you've got a non-standard Go setup then, a quick search for that error mostly shows users who incorrectly built Go from source. We don't have the expertise to help with that, you're best off getting help by raising an issue with the Go team at https://github.com/golang/go/issues.

**Comment by justinvp** [comment]
We're going to want to upgrade gocloud.dev sooner or later (which should help with issues like #14431), but every time we've attempted an upgrade, we've had to revert because it broke functionality for users. Those previous issues could very well be addressed in more recent versions, we'll have to test it out.

In the meantime, we may be able to workaround this by replacing github.com/mitchellh/osext with github.com/kardianos/osext.
